### ISSUE TITLE
Changes to build_artifact_python.sh not reflected when building artifacts

### ISSUE THREAD

**Issue by ghostwriternr** [grpc/grpc#14654]
gRPC version: 1.11.0.dev0 (Built from source)
OS: Ubuntu 16.04 xenial
Kernel: Kernel: x86_64 Linux 4.4.0-112-generic

This is (probably) more of a question and less of an issue. I made some changes to [`build_artifact_python.sh`](https://github.com/grpc/grpc/blob/master/tools/run_tests/artifacts/build_artifact_python.sh) for adding `grpc_testing` to the artifact build process.
Specifically, I started by adding the relevant lines [after the instructions to build the health-checking source distribution](https://github.com/grpc/grpc/blob/59ea0ae3ebcca0aef5a15c5aa5b4d27b7f3fc9c4/tools/run_tests/artifacts/build_artifact_python.sh#L102-L105) as follows:
```sh
...
  # Build gRPC reflection source distribution
  ${SETARCH_CMD} "${PYTHON}" src/python/grpcio_reflection/setup.py \
      preprocess build_package_protos sdist
  cp -r src/python/grpcio_reflection/dist/* "$ARTIFACT_DIR"

  # Build gRPC testing source distribution
  ${SETARCH_CMD} "${PYTHON}" src/python/grpcio_testing/setup.py \
      sdist
  cp -r src/python/grpcio_testing/dist/* "$ARTIFACT_DIR"
fi

cp -r dist/* "$ARTIFACT_DIR"
...
```
But when I run `task_runner.py` as given in `run_tests` README for building artifacts, the change doesn't seem to be reflected at all. It simply skips over the added lines (actually, doesn't see it even).
Command: `tools/run_tests/task_runner.py -f python artifact linux x64`
 
Here's the relevant part of the build logs:
```sh
copying grpcio_reflection.egg-info/requires.txt -> grpcio-reflection-1.11.0.dev0/grpcio_reflection.egg-info
copying grpcio_reflection.egg-info/top_level.txt -> grpcio-reflection-1.11.0.dev0/grpcio_reflection.egg-info
Writing grpcio-reflection-1.11.0.dev0/setup.cfg
creating dist
Creating tar archive
removing 'grpcio-reflection-1.11.0.dev0' (and everything under it)
+ cp -r src/python/grpcio_reflection/dist/grpcio-reflection-1.11.0.dev0.tar.gz /var/local/git/grpc/artifacts/python_linux_x64_cp27-cp27mu
+ cp -r dist/grpcio-1.11.0.dev0.tar.gz /var/local/git/grpc/artifacts/python_linux_x64_cp27-cp27mu
+ cp -r tools/distrib/python/grpcio_tools/dist/grpcio-tools-1.11.0.dev0.tar.gz /var/local/git/grpc/artifacts/python_linux_x64_cp27-cp27mu
+ '[' artifacts '!=' '' ']'
+ docker cp build_and_run_docker_00c32b22-584a-46da-a7b8-6975f1c9b873:/var/local/git/grpc/artifacts /root/grpc
+ docker rm -f build_and_run_docker_00c32b22-584a-46da-a7b8-6975f1c9b873
build_and_run_docker_00c32b22-584a-46da-a7b8-6975f1c9b873
+ '[' '' '!=' '' ']'
PASSED: build_artifact.python_linux_x64_cp27-cp27mu [time=1421.6sec, retries=0:0]
```

I've tried doing this with a fresh git clone & build and a complete docker system prune. Tried running the `task_runner` script both inside a `virtualenv` and out. Been stuck with this for about 2 days now.

Is this expected behaviour or am I missing something? Please help.

**Comment by nathanielmanistaatgoogle** [comment]
Simple sanity-check question, but: have you tried changing the script to do the _wrong_ thing? Like crash? If so, does it crash? If it doesn't, does that mean there's some broken link along the line from code that you know is being run to code that you think should be run but isn't? If so... put crashes in every link in the chain and trace how things are getting run in a way that's different than your expectation?

**Comment by ghostwriternr** [comment]
I did indeed try breaking the file by removing one of the existing build instructions, but the artifacts were getting created nonetheless (i.e. changes not reflected even in that case). I didn't attempt breaking the entire chain though. I'll definitely give it a shot in the morning!

**Comment by nathanielmanistaatgoogle** [comment]
@ghostwriternr: what happened in the morning?

**Comment by ghostwriternr** [comment]
Fixed in #15819. The only extra differences seem to be changes to `MANIFEST.in`.
